File: backend/src/main/java/com/java/NBE4_5_1_7/global/websocket/WebSocketEventListener.java
package com.java.NBE4_5_1_7.global.websocket;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;
import org.springframework.web.socket.messaging.SessionConnectedEvent;
import org.springframework.web.socket.messaging.SessionDisconnectEvent;

@Component
public class WebSocketEventListener {
    private static final Logger logger = LoggerFactory.getLogger(WebSocketEventListener.class);

    @EventListener
    public void handleWebSocketConnectListener(SessionConnectedEvent event) {
        logger.info("✅ WebSocket STOMP 연결 성공: " + event.getUser());
    }

    @EventListener
    public void handleWebSocketDisconnectListener(SessionDisconnectEvent event) {
        logger.info("⛔ WebSocket 연결 종료: " + event.getSessionId());
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/global/websocket/WebSocketEventListener.kt
package com.java.NBE4_5_3_7.global.websocket

import org.slf4j.LoggerFactory
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import org.springframework.web.socket.messaging.SessionConnectedEvent
import org.springframework.web.socket.messaging.SessionDisconnectEvent

@Component
class WebSocketEventListener {

    private val logger = LoggerFactory.getLogger(WebSocketEventListener::class.java)

    @EventListener
    fun handleWebSocketConnectListener(event: SessionConnectedEvent) {
        logger.info("✅ WebSocket STOMP 연결 성공: ${event.user}")
    }

    @EventListener
    fun handleWebSocketDisconnectListener(event: SessionDisconnectEvent) {
        logger.info("⛔ WebSocket 연결 종료: ${event.sessionId}")
    }
}
